### DISCUSSION TITLE
REM/i2c issues

### DISCUSSION THREAD

**Discussion by gw8674** [tagyoureit/nodejs-poolController#863] [REM] [answered]
@rstrouse / @tagyoureit 
I'm having some issues with REM and any assistance you can provide would be greatly appreciated as always...
Here are the particulars.

I currently have pH and ORP Probe Faults indicated in the njspC dashPanel. However, the pH and ORP values indicated are consistent with the values in the REM Web Portal.  I'm also receiving the following errors in the REM CL.
1. error: feedDeviceValue: Could not find i2c device 1:2 - i2c:1:2
2. error: getDeviceStatus: Could not find i2c device 1:2 - i2c:1:2
3. error: getDeviceStatus: Unrecognized I/O Channel null

About 6 days ago, I opened njspC and there were all sorts of errors in the Chemistry section. I began troubleshooting and checked out the REM CL.  Filled with errors. Here were the subsequent steps that I took and the results.

1. I went into device configs and re-saved them and then I checked for free disk space to see if that might be causing corruption but I had 23gb left on the REM pi and 16gb left of the njspC pi.

2. I did a hard power reset and everything went back to normal.

3. Then about 2 days ago, I rechecked everything and things were really bad.  When I opened the REM portal, I had essentially lost the i2c bus/configurations. In the General tab, I2C bus interface and Connections were blank, and there were no I2C bus devices.

4. I had to essentially start from scratch and re-establish the I2C bus, the njspC connection, and all of the Devices in the I2C bus tab.

5. Everything is setup again and is for the most part working normally again (to include auto acid dosing), except for those REM CL errors and the two errors indicated on the njspC dashPanel.

6. I've tried a "Reset Server" in the REM General Tab and also tried both deleting the pH and ORP devices and re-establishing them as well as device resets, but can't seem to get rid of the errors.

Not sure what else to try. Any help would be outstanding...  Up until now, njspC and REM have worked great over the past 3 years and been bulletproof.  Just shooting in the dark here, but I'm wondering if something in the hardware is beginning to fail?  REM pi? The associated Tentacle T3?

Thanks in advance very much for any assistance you can provide!!!

**Comment by gw8674** [comment, marked as answer]
Lo and behold, I just found the problem (at least temporarily anyway)!
On a whim, I was poking around in the njspC dashPanel...
I took another look at the Settings > Chemistry > REM Chem1 > Hardware Tab...
I was missing both pH and ORP selections in the "Device" pulldowns as well as missing checkmarks in both of those "REM Feed" check blocks.

So I made those corrections and sure enough the two errors in the njspC Chemistry block as well as the three REM CL errors "magically" disappeared. Other than the occasional "suspend polling" warnings in the CL, everything there looks good.

**Note-to-Self**: Check **EVERYTHING** first before throwing in the towel and yelling for help. :)

Sorry for the false alarm....  Still curious why my I2C bus/REM got discombobulated in the first place. Y'all probably have some theories but I'm just not sure.  I just hope I get another 3 years before another occurrence like this.

As always though, thanks for listening!!!
r.

**Reply by rstrouse** [reply]
You can have a condition where the hardware tab does not fill in the data if REM has not had time to finish its initial run through of its hardware on the I2c bus and the controller is loaded.   This is typically not a problem unless you are restarting REM and decide to edit the REM Chem controller while it is restarting.  Like if you were restarting servers to troubleshoot a problem.  

When it queries REM it will return all the devices that it knows about at the time but during restart that list will not yet be complete as each I2c communication has to be done sequentially.

**Reply by gw8674** [reply]
Thanks @rstrouse  Makes sense now.  That explains why I lost the configurations in the njspC dashPanel Settings > Chemistry > REM Chem1 > Hardware Tab...

As far as what happened to REM and the I2C bus originally, who knows?  I'm inclined to just concede to the "gremlin factor" (hardware?) and move on. I've been keeping a real close eye on the REM portal, REM CL and njspC dashPanel and so far, everything remains high and tight.

Thanks again, take care and have an enjoyable turkey day!!
